modules/tinymce/src/plugins/table/main/ts/core/TableSections.ts

```diff
@@ -80,11 +81,27 @@ const switchRowSection = (dom: DOMUtils, rowElm: HTMLElement, newSectionName: st
   }
 };
 
-const switchCellType = (dom: DOMUtils, cells: ArrayLike<HTMLTableCellElement>, newCellType: string, scope: 'col' | null) =>
-  Arr.each(cells, (c) => {
-    const newCell = Util.getNodeName(c) !== newCellType ? dom.rename(c, newCellType) : c;
+const renameCell = (editor: Editor, cell: HTMLTableCellElement, newCellType?: string): HTMLTableCellElement => {
+  if (Type.isNonNullable(newCellType) && Util.getNodeName(cell) !== newCellType) {
+    const newCellElm = editor.dom.rename(cell, newCellType) as HTMLTableCellElement;
+    Events.fireNewCell(editor, newCellElm);
+    return newCellElm;
+  } else {
+    return cell;
+  }
+};
+
+const switchCellType = (editor: Editor, cell: HTMLTableCellElement, newCellType?: string, scope?: 'col' | null): HTMLTableCellElement => {
+  const dom = editor.dom;
+  const newCell = renameCell(editor, cell, newCellType);
+  if (!Type.isUndefined(scope)) {
```

Comment:
Just wanted to sanity check, the `newCell` event will fire before the scope is possibly set right?

Reply:
There are some other cases where there will be modifications that will happen after `newCell` too (e.g. styling). The `TableModified` event will be fired after this entire process is complete, should an integrator wish to wait for everything to finish, so this should be fine.